{
  try {
    realRun();
    threadShouldThrow("InterruptedException");
  }
 catch (  InterruptedException success) {
  }
catch (  Throwable t) {
    threadUnexpectedException(t);
  }
}
